core/Tracker/Visit/VisitProperties.php

```diff
@@ -0,0 +1,58 @@
+<?php
+/**
+ * Piwik - free/libre analytics platform
+ *
+ * @link http://piwik.org
+ * @license http://www.gnu.org/licenses/gpl-3.0.html GPL v3 or later
+ */
+
+namespace Piwik\Tracker\Visit;
+
+/**
+ * Holds temporary data for tracking requests.
+ *
+ * RequestProcessors
+ */
+class VisitProperties
+{
+    /**
+     * Information about the current visit. This array holds the column values that will be inserted or updated
+     * in the `log_visit` table, or the values for the last known visit of the current visitor.
+     *
+     * @var array
+     */
+    public $visitorInfo = array();
+
+    /**
+     * Stores plugin specific tracking request metadata. RequestProcessors can store
+     * whatever they want in this array, and other RequestProcessors can modify these
+     * values to change tracker behavior.
+     *
+     * @var string[][]
+     */
+    private $requestMetadata = array();
+
+    /**
+     * Set a request metadata value.
+     *
+     * @param string $pluginName eg, `'Actions'`, `'Goals'`, `'YourPlugin'`
+     * @param string $key
+     * @param mixed $value
+     */
+    public function setRequestMetadata($pluginName, $key, $value)
+    {
+        $this->requestMetadata[$pluginName][$key] = $value;
```

Comment:
Why should it be in core? Would it be unexpected for a plugin to change it? The logic that sets it isn't in core.

It can exist in a class instead of as an element in an array, see below. W/o support for generics, it will never look perfect.